pkg/declared/scope.go

```diff
@@ -41,3 +45,36 @@ func ValidateScope(s string) error {
 	}
 	return nil
 }
+
+// ScopeFromSync returns the scope associated with a specific RootSync or RepoSync.
+func ScopeFromSync(syncKind, syncNamespace string) Scope {
```

Comment:
`ScopeFromSync` can be simplified as `ScopeFromSyncNamspace` with only one parameter because RepoSync can't be defined in the c-m-s namespace, while RootSync can only exist in the c-m-s namespace.

Reply:
I've never heard of this constraint. Do you know where the validation for it is?

Reply:
Found it https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/reposync_controller.go#L945

Reply:
Wish I'd known that before. That's useful for all sorts of naming length reductions. You can just use NN, without needing to specify the kind.

Reply:
Sorry, missed that. 

And https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/rootsync_controller.go#L823 for RootSync.

I think having kind in the labels is more explicit. For naming length reduction, yes, we can strip out the kind part.

Reply:
Ok. I refactored the helpers a bit and made a couple of them methods on the Scope type.